code/datums/components/chameleon/spritesheet.dm

```diff
@@ -0,0 +1,42 @@
+/datum/asset/spritesheet/chameleon
+	name = "chameleon"
+
+/datum/asset/spritesheet/chameleon/register()
+	var/list/disguises = list()
+	// First, get everything we need to make icons of.
+	for(var/chameleon_path in subtypesof(/datum/component/chameleon))
+		var/datum/component/chameleon/chameleon = chameleon_path
+		if(!initial(chameleon.base_disguise_path))
+			continue
+		disguises |= list_chameleon_disguises(initial(chameleon.base_disguise_path), initial(chameleon.disguise_blacklist), initial(chameleon.hide_duplicates))
+	// Then, we need to generate the actual icons.
+	for(var/item_path in disguises)
+		add_item(item_path)
+	// I like this better than hardcoding a list of all chameleon items.
+	for(var/item_path in subtypesof(/obj/item))
+		var/item_path_txt = "[item_path]"
+		if(!findtext(item_path_txt, "/chameleon") || findtext(item_path_txt, "/broken") || (item_path in disguises))
```

Comment:
Bllleeeeeeeeeeeeeeeeech

Reply:
i can't think of a better way besides hardcoding

Reply:
How was it done before

if it was hardcoded before tbh just do that. There's not really an insane amount of chameleon items

Reply:
> How was it done before

It... _wasn't_ doing this before. This is in _spritesheet generation code_. There were no sprites to generate before.